src/runtime/recipe/recipe-resolver.ts

```diff
@@ -112,7 +133,7 @@ export class ResolveWalker extends RecipeWalker {
 
      // SlotUtils handles a multi-slot case.
     if (allSlots.length !== 1) {
-      return undefined;
+      return error('There is not just one slot');
```

Comment:
'There are multiple matching slots' ?

Reply:
Added `(match is ambiguous)` to your suggestion to make it clear why this is a problem (though I think we probably should just try each option, though there is a combinatorial explosion).

Reply:
Combinatorial explosion is dealt with if you use the Strategizer, RecipeResolver is explicitly intended to only deal with cases where there's 1-1 mapping and no explosion. These 2 share some code, so it can be interesting, but in your case it's clear what to do, because you're modifying RecipeResolver here.

Reply:
Ah, thanks. I hadn't realized that the two approaches were so neatly separated.